internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -123,7 +129,71 @@ func (r *NonAdminBackupStorageLocationReconciler) Reconcile(ctx context.Context,
 		}
 	}
 
-	logger.V(1).Info("NonAdminBackup Reconcile exit")
+	if previousBackupSyncRun == nil || time.Now().After(previousBackupSyncRun.Add(r.SyncPeriod)) {
```

Comment:
Why not introduce a separate controller to sync the backups ? 
Also, if not a new controller then would it make sense to add this in NAB controller instead of NABSL ?

Reply:
was to make logic simpler

with decision to use default BSLs from OADP namespace, its not feasible anymore